src/System.Management.Automation/engine/Utils.cs

```diff
@@ -594,6 +594,47 @@ internal static string GetCurrentMajorVersion()
             return PSVersionInfo.PSVersion.Major.ToString(CultureInfo.InvariantCulture);
         }
 
+        /// <summary>
+        /// Checks whether the user's OS type is in a list of required types.
+        /// </summary>
+        /// <param name="requiredOSTypes">A list of required types.</param>
+        /// <returns>True if the user's OS type is in the list of required types, false otherwise.</returns>
+        internal static bool IsOSTypeValid(IEnumerable<string> requiredOSTypes)
+        {
+            string currentOSName = GetOSTypeString();
+            foreach (string requiredOSType in requiredOSTypes) 
+            {
+                if (requiredOSType.Equals(currentOSName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
```

Comment:
Not sure if LINQ is what we want to use here, but it would simplify the code to use the .Contains() method:
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.contains?view=netframework-4.8#System_Linq_Enumerable_Contains__1_System_Collections_Generic_IEnumerable___0____0_System_Collections_Generic_IEqualityComparer___0__